### PR TITLE
Tt 26 warn of unsaved changes on customer page

### PR DESCRIPTION
**Pull request by scastillo-jp**

### CODE REVIEW

File: src/app/modules/customer-management/components/customer-info/components/customer-list/customer-list.component.ts [hunk 1]
```diff
@@ -77,10 +79,25 @@ export class CustomerListComponent implements OnInit, OnDestroy, AfterViewInit {
   }
 
   editCustomer(customerId: string) {
+    this.idToEdit = customerId;
+    if (this.haveChanges === true) {
```
Comment (thegreatyamori) [inline src/app/modules/customer-management/components/customer-info/components/customer-list/customer-list.component.ts:83]: Does the comparison need to be explicit?
`this.haveChanges === true`

File: src/app/modules/customer-management/components/projects-type/components/create-project-type/create-project-type.component.ts [hunk 2]
```diff
@@ -63,9 +65,11 @@ export class CreateProjectTypeComponent implements OnInit, OnDestroy {
         id: this.projectTypeToEdit.id,
       };
       this.store.dispatch(new UpdateProjectType(projectType));
+      this.isHaveChanges.emit(this.haveChanges = false);
     } else {
       this.store.dispatch(new CreateProjectType({ ...projectTypeData, customer_id: this.customerId }));
       this.projectTypeForm.get('description').setValue('');
+      this.isHaveChanges.emit(this.haveChanges = false);
```
Comment (thegreatyamori) [inline src/app/modules/customer-management/components/projects-type/components/create-project-type/create-project-type.component.ts:72]: The `obSubmit` function always output a `this.haveChanges = false`?

File: src/app/modules/customer-management/components/projects-type/components/create-project-type/create-project-type.component.ts [hunk 3]
```diff
@@ -76,4 +80,9 @@ export class CreateProjectTypeComponent implements OnInit, OnDestroy {
   ngOnDestroy(): void {
     this.getCustomerIdSubscription.unsubscribe();
   }
+
+  onSearchChanges(searchValue: string): void {
```
Comment (thegreatyamori) [inline src/app/modules/customer-management/components/projects-type/components/create-project-type/create-project-type.component.ts:84]: This function is identical to `onSearchChanges` of create-customer.ts
[Here](https://github.com/ioet/time-tracker-ui/blob/6d8b0b5003c99bb4f607f25c902af726e52b3c3c/src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts#L101)

File: src/app/modules/customer-management/components/projects/components/create-project/create-project.component.ts [hunk 4]
```diff
@@ -107,4 +109,9 @@ export class CreateProjectComponent implements OnInit, OnDestroy {
       project_type_id: null,
     });
   }
+
+  onSearchChanges(searchValue: string): void {
```
Comment (thegreatyamori) [inline src/app/modules/customer-management/components/projects/components/create-project/create-project.component.ts:113]: This function is identical to `onSearchChanges` of create-customer.ts
[Here](https://github.com/ioet/time-tracker-ui/blob/6d8b0b5003c99bb4f607f25c902af726e52b3c3c/src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts#L101)

File: src/app/modules/customer-management/components/customer-info/components/customer-list/customer-list.component.ts (hunk 1 above)
Comment (scastillo-jp) [reply src/app/modules/customer-management/components/customer-info/components/customer-list/customer-list.component.ts:83]: I think it is not necessary I will check it thanks

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.html [hunk 5]
```diff
@@ -23,6 +24,7 @@
         rows="3"
         formControlName="description"
         placeholder="Customer description"
+        (input)="onSearchChanges($event.target.value)"
```
Comment (Angeluz-07) [inline src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.html:27]: I have a doubt. Why is the function called `onSearchChanges`? I remember the purpose is to check when the value Input field has changed. Maybe we need better name like `onInputChange` or something similar.

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts [hunk 6]
```diff
@@ -23,6 +23,8 @@ import { LoadCustomerProjects, CleanCustomerProjects } from '../../../projects/c
 export class CreateCustomerComponent implements OnInit, OnDestroy {
   customerForm: FormGroup;
   @Input() areTabsActive: boolean;
+  @Input() haveChanges: boolean;
```
Comment (Angeluz-07) [inline src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts:26]: I think you meant `hasChanges` or `hasChanged`.

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts [hunk 7]
```diff
@@ -23,6 +23,8 @@ import { LoadCustomerProjects, CleanCustomerProjects } from '../../../projects/c
...
+  @Output() isHaveChanges = new EventEmitter<boolean>();
```
Comment (Angeluz-07) [inline src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts:27]: Maybe a better name would be `hasChangedEvent` or `inputHasChangedEvent`.

File: src/app/modules/customer-management/components/projects-type/components/create-project-type/create-project-type.component.ts (hunk 2 above)
Comment (scastillo-jp) [reply src/app/modules/customer-management/components/projects-type/components/create-project-type/create-project-type.component.ts:72]: I checked this line, is no necessary.

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.html (hunk 5 above)
Comment (scastillo-jp) [reply src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.html:27]: Yes, the purpose is to check when the `Input` value or the `textArea` has changed.
I'm going to rename `onSearchChanges`

**Review by thegreatyamori — APPROVED** [review]

**Review by Angeluz-07 — APPROVED** [review]

**Review by kellycastrof — APPROVED** [review]

**Pull request merged**